modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp

```diff
@@ -1103,6 +1101,30 @@ struct VST3ModuleHandle  : public ReferenceCountedObject
         return false;
     }
 
+    Result findName(IPluginFactory* factory, const PluginDescription& description)
+    {
+        if (factory == nullptr)
+            return Result::fail(L"Invalid Parameter");
+
+        Result result(Result::fail(L"not found"));
+        auto numClasses = factory->countClasses();
+
+        for (Steinberg::int32 i = 0; i < numClasses; ++i)
+        {
+            PClassInfo info;
+            factory->getClassInfo(i, &info);
+
+            if (std::strcmp(info.category, kVstAudioEffectClass) != 0)

```

Comment:
I vaguely recall that many audio plugins don't support this method... I suggest testing with a variety of open-source plugs to be sure.

Reply:
Have a look at `DescriptionFactory` to get a better sense of it.

Reply:
I picked this piece of code from DescriptionFactory. Basically this findName() method I added is the same as DescriptionFactory::findDescriptionsAndPerform() but I removed anything that is not needed. So the criteria are the same as before except we don't instantiate and destroy the component.
What I want to check in this loop is: 
  - the name of the component (does it match the the provided one nested in the passed description)
  - is it an audio processor (we ignore the user interface component)